example/test/function_component_test.dart

```diff
@@ -37,29 +33,96 @@ UseCallbackTestComponent(Map props) {
   }, []);
 
   return react.div({}, [
-    react.div({}, ['Delta is ${delta.value}']),
-    react.div({}, ['Count is ${count.value}']),
-    react.button({'onClick': increment}, ['Increment count']),
-    react.button({'onClick': incrementDelta}, ['Increment delta']),
+    react.div({'key': 'ucbt1'}, ['Delta is ${delta.value}']),
+    react.div({'key': 'ucbt2'}, ['Count is ${count.value}']),
+    react.button({'onClick': increment, 'key': 'ucbt3'}, ['Increment count']),
+    react.button({'onClick': incrementDelta, 'key': 'ucbt4'}, ['Increment delta']),
+  ]);
+}
+
+var useContextTestFunctionComponent =
+    react.registerFunctionComponent(UseContextTestComponent, displayName: 'useContextTest');
+
+UseContextTestComponent(Map props) {
+  final context = useContext(TestNewContext);
```

Comment:
It bothers me a bit that `context` will be dynamic unless you provide a type on the left side, and even then - there'd be no way to ensure that the type doesn't mismatch with the underlying type that was returned by `createContext`.  Makes me wonder what the point of the generic parameter is on `createContext`.

By making these changes locally... all the typing seems to line up and makes `context` correctly inferred as a `Map` in this situation as a result of `createContext<Map>` being called on line 63.  

```diff
// context.dart

- class Context {
+ class Context<T> {
// ...
}

- Context createContext<TValue>([
+ Context<TValue> createContext<TValue>([
// ...
```

```diff
// hooks.dart

- dynamic useContext(Context context) => // ...
+ T useContext<T>(Context<T> context) => // ...
```

@greglittlefield-wf @kealjones-wk @joebingham-wk is there some reason that I forgot about that led us to omit generic typing from the `Context` class?

Reply:
Honestly I don't recall, I feel like there would have been a reason because the `Context` object in OverReact is typed exactly like that and i don't know why i would have omitted it in React-Dart unless it was just a minor oversight (very likely). @joebingham-wk if you can add this typing id say do it, but test the local context examples in browser and test it with over_react (tests & real examples if already in place) to make sure it doesn't break anything (i don't think it should).